### PR TITLE
Make doctests work with Python 2/3 string outputs

### LINKED ISSUE #87: Turn 'how to create a new dataset' tutorial into unit tests

### PR DESCRIPTION
**Pull request by vdumoulin**
This allows to remove a bunch of `doctest: +SKIP` statements and improve overall test coverage.

Fixes #87.

### CODE REVIEW

**Comment by bartvm** [comment]
I'd prefer it the other way around, actually, because the documentation is supposed to be Python 3. I guess that applying the regular expression to `got` instead of `want` would allow you to make everything work too, without the need to put `'u` in the docstrings everywhere?

File: doctests/__init__.py [hunk 1]
```diff
@@ -5,11 +5,31 @@
...
+    .. _original blog post:
+       https://dirkjan.ochtman.nl/writing/2014/07/06/
+       single-source-python-23-doctests.html
+
+    """
+    def check_output(self, want, got, optionflags):
+        if sys.version_info[0] > 2:
+            want = re.sub("u'(.*?)'", "'\\1'", want)
+            want = re.sub('u"(.*?)"', '"\\1"', want)
+        return doctest.OutputChecker.check_output(self, want, got, optionflags)
```
Comment (bartvm) [inline doctests/__init__.py]: Why not `OutputChecker.check_output` (it's imported after all) or, even better, `super(Py23DocChecker, self).check_output(want, got, optionflags)`

Comment (vdumoulin) [inline doctests/__init__.py]: It's just a copy-paste from the blog post, but sure, I can do that.

Comment (vdumoulin) [inline doctests/__init__.py]: I'll have to settle for `Outpuchecker.check_output`, as `OutputChecker` is an old-style class and subclassing it prevents me from using `super`.

**Comment by bartvm** [comment]
Cool, LGTM.

Too bad that you need to duplicate all the `code-block`s into hidden doctests though. Maybe we can make a ticket for it, and implement a custom `DocTestParser` sometime in the future. It should be possible to basically add a `:doctest:` directive to `code-block`s, that will be picked up by a custom `DocTestParser`.

**Comment by vdumoulin** [comment]
Yeah, it kind of bothers me as well. I think I might have something for that, though, let met give it a try.

**Comment by vdumoulin** [comment]
Nevermind, looks like `doctest` only recognizes interpreter-like commands.

**Comment by vdumoulin** [comment]
Shall I go ahead and merge?

**Comment by bartvm** [comment]
Yeah, go ahead.

**Pull request merged**